docs/src/main/sphinx/language/types.rst

```diff
@@ -5,10 +5,32 @@ Data types
 Trino has a set of built-in data types, described below.
 Additional types can be provided by plugins.
 
-.. note::
+.. _type-mapping-overview:
+
+Trino type support and mapping
+------------------------------
+
+Connectors to data sources are not required to support all Trino data types
+described on this page. If there are data types similar to Trino's that are used
+on the data source, the connector may map the Trino and remote data types to
+each other as needed.
+
+Depending on the connector and the data source, type mapping may apply
+in either direction as follows:
+
+* **Data source to Trino** mapping applies to read and write operations, when
+  columns with specific types in the data source exist and are read or written
+  from Trino. In this type mapping table, all supported types in the data
+  source should be listed. Multiple remote data types may map to the same Trino
+  data type.
+
+* **Trino to data source** mapping applies to any SQL statements that create
+  columns on the data source (such as :doc:`/sql/create-table` and
+  :doc:`/sql/alter-table`). In this type mapping table, all supported data types
```

Comment:
I think we also need to mention write operations (e.g. `INSERT`, `CREATE TABLE AS SELECT` etc.)

Reply:
sure.. but they are covered by the other direction of mapping... .. the Trino to data source mapping is essentially for DDL statements only..

Reply:
Got it, thanks for the explanation.

Reply:
Small clarification (to be pedantic) - Trino to data source mapping is used in write operations as well as things like predicate pushdown - basically anywhere we need to convert from Trino's representation to the data source's representation.

Reply:
No change requested - was just a clarification.